indexer/map_object.cpp

```diff
@@ -94,6 +94,59 @@ string MapObject::GetLocalizedType() const
   return platform::GetLocalizedTypeName(classif().GetReadableObjectName(copy.GetBestType()));
 }
 
+std::string MapObject::GetAllLocalizedTypes() const
+{
+  ASSERT(!m_types.Empty(), ());
+  feature::TypesHolder copy(m_types);
+  copy.SortBySpec();
+
+  auto const & isPoi = ftypes::IsPoiChecker::Instance();
+  auto const & isGeneralAmenity = ftypes::IsGeneralAmenityChecker::Instance();
+
+  std::ostringstream oss;
+  bool isFirst = true;
+  for (auto const type : copy)
+  {
+    // Ignore types that are not POI
+    // Ignore general "amenity" type that has no subtype
+    // But always show the first/main type
+    if (!isFirst && (!isPoi(type) || isGeneralAmenity(type)))
```

Comment:
Better ```(!isPoint(type) || ftype::GetLevel(type) == 1)``` and remove confusing IsGeneralAmenityChecker class.

Reply:
Okay! Makes sense! Thanks!